src/witan/send/adroddiad/ncy.clj

```diff
@@ -1,7 +1,298 @@
+
+  Ages and NCYs for children aged 4 to 15 at the start of school/academic
+  year (reception to NCY 11) are per https://www.gov.uk/national-curriculum.
+  Extension to ages 0-3 (NCYs -4 to -1) and ages 16-24 (NCYs 12 to 20),
+  is by linear extrapolation, such that the offset between NCY and age
+  at the beginning of the school/academic year is -4.
+
+  Age at the start of the school/academic year is the age on 31st August
+  prior to the school/academic year. This is per:
+
+  - The [gov.uk website](https://www.gov.uk/schools-admissions/school-starting-age),
+  which (as of 23-NOV-2022) states:
+  \"Most children start school full-time in the September after their
+  fourth birthday. This means they’ll turn 5 during their first school
+  year. For example, if your child’s fourth birthday is between 1
+  September 2021 and 31 August 2022 they will usually start school in
+  September 2022.\".
+
+  - The [2022 SEN2 guide](https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/1013751/SEN2_2022_Guide.pdf),
+  which states in section 2, part 1, item 1.1 that: \"age breakdown
+  refers to age as at 31 August 2021\", implying (since this is for
+  the January 2022 SEN2 return) that age for SEN2 breakdowns is as of
+  31-AUG prior to starting the school year.
+
+  The maximum age for SEND is 25, but per the relevant legistlation
+  \"a local authority may continue to maintain an EHC plan for a young
+  person until the end of the academic year during which the young
+  person attains the age of 25\", such that they will have been 24 at
+  the start of that school/academic year, hence the maximum age in
+  this map is 24 (NCY 20).
+  "
+  (set/map-invert ncy->age-at-start-of-school-year))
+
+
+
+(rcf/tests                              ; tests for ncy->age… and age…->ncy
+
+ "Age at start of reception should be 4 years, per gov.uk/schools-admissions/school-starting-age"
+ (ncy->age-at-start-of-school-year 0) := 4
+
+ "Age at start of NCY -4 to 20 should be 0 to 24 respectively"
+ (map ncy->age-at-start-of-school-year (range -4 (inc 20))) := (range 0  (inc 24))
+
+ "ncy->age… should return nil for non-integer NCYs outside -4 to 20 inclusive"
+ (map ncy->age-at-start-of-school-year [-5 0.5 21]) := '(nil nil nil)
+
+ "NCY entered at age 4 should be 0 (reception), per gov.uk/schools-admissions/school-starting-age"
+ (age-at-start-of-school-year->ncy 4) := 0
+
+ "NCY entered at age 0 to 24 should be -4 to 20 respectively"
+ (map age-at-start-of-school-year->ncy (range  0 (inc 24))) := (range -4 (inc 20))
+ 
+ "Round trip test: ncy->age… followed by age…->ncy should return the same for integers -4 to 20 inclusive"
+ (map (comp age-at-start-of-school-year->ncy ncy->age-at-start-of-school-year) (range -4 (inc 20))) := (range -4 (inc 20))
+ 
+ "age…->ncy should return nil for non-integer age outside 0 to 24 inclusive"
+ (map age-at-start-of-school-year->ncy [-1 0.5 25]) := '(nil nil nil)
+ 
+ )
```

Comment:
Good to see the tests. I do wonder about having them all at the bottom as that is where I'd expect rich comment blocks and the like to be. Not a big issue with a smaller namespace.